tools/SourceKit/tools/complete-test/complete-test.cpp

```diff
@@ -21,14 +21,26 @@
 #include "llvm/Support/FileSystem.h"
 #include <fstream>
 #include <regex>
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
 #include <unistd.h>
 #include <sys/param.h>
+#elif defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#include <Windows.h>
+#endif
 
 // FIXME: Platform compatibility.
 #include <dispatch/dispatch.h>
 
 using namespace llvm;
 
+#if defined(_WIN32)
+namespace {
+int STDOUT_FILENO = _fileno(stdout);
```

Comment:
We generally try to avoid running code for global variables because it incurs code that always runs at initialization, even if it's not used. I understand this is only for a testing utility but it would be good to be consistent across the codebase. My recommendations would be:
* It is plausible that this is not the first time across all of llvm/clang/swift that code needed to something similar, see if there's an API somewhere that already exists for getting the stdout file number in a platform-independent way.
* If you can't find anything available, add an API in 'swift/Basic' so it can be re-used.
* The API should be a function call, not a global variable.